# Common configuration for PX4 autopilot

# node:
startup_px4_usb_quirk: true

##
## sys_time
##

conn:
  timesync_rate: 10.0     # TIMESYNC rate in Hertz (feature disabled if 0.0)
  system_time_rate: 1.0   # send system time to FCU rate in Hertz (disabled if 0.0)
time:
  time_ref_source: "fcu"  # time_reference source
  timesync_avg_alpha: 0.6 # timesync averaging factor
  max_rtt_sample: 50.0    # for timesync: samples with round-trip time higher than max_rtt_sample are not used to update the filter
  timesync_mode: MAVLINK  # if the fcu and companion computer should synchronize and if so how

##
## sys_status
##

conn:
  timeout: 10.0          # connection timeout in seconds
  heartbeat_rate: 1.0    # send HEARTBEAT message rate [Hz] (disabled if 0.0)
sys:
  min_voltage: 10.0      # minimal battery voltage for diagnostics
  disable_diag: false    # disable all diagnostics except HEARTBEAT message

##
## command
##

cmd:
  use_comp_id_system_control: false # quirk for some old FCUs

##
## imu_pub
##

imu:
  # frame_id: "base_link" set in launch file TODO should technically be fcu?
  linear_acceleration_stdev: 0.0003     # default 0.0003
  angular_velocity_stdev: !degrees 0.02 # default 0.02
  orientation_stdev: 1.0                # default 1.0
  magnetic_stdev: 0.0                   # default 0.0

##
## global_position
##

global_position:
  frame_id: "world"             # origin frame
  #child_frame_id: "base_link"  # body-fixed frame, set in launch file
  rot_covariance: 99999.0       # covariance for attitude?
  use_relative_alt: true        # use relative altitude for local coordinates
  tf:
    send: false                 # send TF?
    frame_id: "tf_frame_id"    # TF frame_id
    global_frame_id: "tf_global_frame_id"    # TF earth frame_id
    child_frame_id: "tf_child_frame_id" # TF child_frame_id

##
## local_position
##

local_position:
  #frame_id: "map" # set in launch file
  tf:
    send: false
    # frame_id: "aero2/map" # set in launch file
    # child_frame_id: "aero2/base_link" # set in launch file
    send_fcu: false

##
## setpoint_position
##

setpoint_position:
  tf:
    listen: false           # enable tf listener (disable topic subscribers)
    frame_id: "map"
    child_frame_id: "target_position"
    rate_limit: 50.0
  mav_frame: LOCAL_NED

##
## setpoint_velocity
##

setpoint_velocity:
  mav_frame: LOCAL_NED
